index.rst

```diff
@@ -53,10 +53,362 @@
+and
+https://github.com/lsst-dm/sample-avro-alert.
+
+To build the images for the containers for the alert stream producers,
+filters, and consumers, checkout the ``alert_stream`` repo.
+From the ``alert_stream`` directory:
+
+.. code-block:: python
+
+  docker build -t "alert_stream" .
+
+
+Kafka
+-------------
+
+This prototype uses Confluent's distribution of Kafka, specifically
+Confluent 4.1.1 and Kafka 1.1.0.
+Kafka and Zookeeper can be run in containers.
+To start Kafka and Zookeeper on a single node using
+Docker Compose (i.e., not Swarm mode),
+from the ``alert_stream`` directory run:
+
+.. code-block:: python
+
+  docker-compose up -d
+
+In a larger scale Swarm, initiate the Swarm across
+all machines and run the commands below to start the services.
+It is good to keep Kafka and Zookeeper separate from
+other containers by constraining Kafka and Zookeeper to the manager node(s)
+and constraining the other services to worker nodes.
+
+.. code-block:: python
+
+  docker network create --driver overlay alert_stream_default
+
+  docker service create \
+          --name kafka \
+          --network alert_stream_default \
+          --constraint node.role==manager \
+          -p 9092 \
+          -e KAFKA_BROKER_ID=1 \
+          -e KAFKA_ZOOKEEPER_CONNECT=zookeeper:32181 \
+          -e KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://kafka:9092 \
+          -e KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 \ # remove if starting 3 brokers or more
+          -e KAFKA_HEAP_OPTS="-Xmx8g -Xms8g" \
+          -e KAFKA_JVM_PERFORMANCE_OPTS="-XX:MetaspaceSize=96m -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35 -XX:G1HeapRegionSize=16M -XX:MinMetaspaceFreeRatio=50 -XX:MaxMetaspaceFreeRatio=80" \
+          confluentinc/cp-kafka:4.1.1
+
+  docker service create \
+          --name zookeeper \
+          --network alert_stream_default \
+          --constraint node.role==manager \
+          -p 32181 \
+          -e ZOOKEEPER_CLIENT_PORT=32181 \
+          -e ZOOKEEPER_TICK_TIME=2000 \
+          confluentinc/cp-zookeeper:4.1.1
+
+Kafka and Zookeeper should be listed when running
+
+.. code-block:: python
+
+  docker service ls
+
+
+End consumers
+-------------
+
+Sample consumers can be started by running either ``printStream.py``,
+which prints alert contents to the screen,
+or ``monitorStream.py``, which prints the status of the latest offset
+(number of alerts received).
+Both of these scripts are found in the ``alert_stream`` GitHub rep.
+
+To run the mini-broker prototype at full scale with 100 end-users,
+100 containers would need to be deployed, each consuming a topic
+following the naming scheme
+``Filter001``, ``Filter002``, etc... ``Filter100``.
+
+To deploy, e.g., the monitoring script using Docker on a single node,
+consuming the stream for the 10th filter, run:
+
+.. code-block:: python
+
+  docker run -it --rm \
+             --name=monitor010 \
+             --network=alert_stream_default \
```

Comment:
note that `docker-compose up -d` creates a network `alert_stream_lsst_default` (added `_lsst`)--I would harmonize this throughout the document.

Reply:
Do you mean to say that you want me to change the network name or that you observed that _lsst is added to the network name?  I used that name because the network name should be autocreated from the directory name, which is alert_stream.  Is that not true for you?

Reply:
Ah, I see--problem on my side, then (I have another `alert_stream` directory)...